dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,47 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+function getSummarizedProgressForAnonymousUser () {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';');
```

Comment:
Please use a well-tested JS cookie handling library (hopefully we have one already) rather than doing inline cookie cracking which can be error-prone.